speccyMac/Loaders/Z80Loader.swift

```diff
@@ -16,6 +40,144 @@ class Z80Loader: GameLoaderProtocol {
+        v1.sp = UInt16(data[8]) + (UInt16(data[9]) << 8)
+        v1.i = data[10]
+        v1.r = data[11]
+        v1.flags1 = data[12]
+        v1.de = UInt16(data[13]) + (UInt16(data[14]) << 8)
+        v1.exbc = UInt16(data[15]) + (UInt16(data[16]) << 8)
+        v1.exde = UInt16(data[17]) + (UInt16(data[18]) << 8)
+        v1.exhl = UInt16(data[19]) + (UInt16(data[20]) << 8)
+        v1.exa = data[21]
+        v1.exf = data[22]
+        v1.iy = UInt16(data[23]) + (UInt16(data[24]) << 8)
+        v1.ix = UInt16(data[25]) + (UInt16(data[26]) << 8)
+        v1.iff = data[27]
+        v1.iff2 = data[28]
+        v1.flags2 = data[29]
+
+        // file format quirks
+        v1.r = (v1.r & 0x7f) | (v1.flags1 << 7)
+        if v1.flags1 == 255 { v1.flags1 = 1 }
+
+        if (v1.pc != 0) {
+            print(".z80 version 1 file")
+
+            // compressed files should never be longer than compressed, which are 49182 bytes
+            if data.count > (48 * 1024) + v1HeaderLen {
+                print(" file too long")
+                return false
+            }
+
+            let isCompressed = v1.flags1 & (1 << 5) != 0
+            print("  \(!isCompressed ? "un" : "")compressed data")
+
+
+            // prepare CPU and hardware
+
+            z80.i = v1.i
+
+            z80.exhl = v1.exhl
+            z80.exde = v1.exde
+            z80.exbc = v1.exbc
+            z80.exaf = (UInt16(v1.exa) << 8) + UInt16(v1.exf)
+
+            z80.hl.value = v1.hl
+            z80.de.value = v1.de
+            z80.bc.value = v1.bc
+            z80.iy = v1.iy
+            z80.ix = v1.ix
+
+            // z80[27] =  sna[19].00001000
+            if v1.iff != 0 {
+                z80.interrupts = true
+                z80.iff1 = 1
+                z80.iff2 = 1
+            } else {
+                z80.interrupts = false
+                z80.iff1 = 0
+                z80.iff2 = 0
+            }
+
+            z80.r.value = v1.r
+
+            z80.af.value = (UInt16(v1.a) << 8) + UInt16(v1.f)
+            ZilogZ80.sp = v1.sp
+
+            // sna[25].00000011 = z80[29].00000011
+            z80.interruptMode = v1.flags2 & 0x03
+            // sna[26].00000111 = z80[12].00001110
+            z80.machine?.output(0xfe, byte: (v1.flags1 >> 1) & 0x07)
+
+            // prepare ram
+
+            let ram = z80.memory.romSize
+            let ramlen = 64 * 1024 // can change if 16k support is added
+            let datlen: UInt16 = UInt16(data.count) - v1HeaderLen
+            var i: UInt16 = v1HeaderLen
+            var o: UInt16 = 0
+
+            var loop = true
+            while loop {
+                let left = datlen - i // TODO can overflow?
+                if left == 0 {
+                    print("    end of data")
+                    break
+                }
+                if UInt(ram) + UInt(o) >= ramlen {
+                    print("    end of ram")
+                    return false
+                }
+
+                let b = data[Int(i)]
```

Comment:
Please can we avoid 1 letter variable names where possible (obviously this doesn't include z80 vars)